Assets/Scripts/MeleeController.cs

```diff
@@ -0,0 +1,34 @@
+using UnityEngine;
+
+public class MeleeController : MonoBehaviour
```

Comment:
Maybe not needed, but it is good to check things that have common behaviour, common properties and define them an interface or base class, this way we could avoid duplicate code and maybe in the future reuse some parts. 
What I mean is, instead of coding same thing twice, we can code it once.